### PR TITLE
Issue #11: provide UT for path on multiple files

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #11: provide UT for path on multiple files

### CODE REVIEW

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java [hunk 1]
```diff
@@ -92,25 +97,69 @@ protected void finishLocalSetup() throws Exception {
...
+     * @return patchedList List of List of String
      */
-    private String getFileName(List<String> patched) {
-        return patched.get(1).split("\\s")[1];
+    private List<List<String>> getPatchList(List<String> originPatch) {
+        final List<List<String>> patchedList = new ArrayList<>();
+        boolean flag = true;
+        List<String> singlePatched = new ArrayList<>();
+        for (String str : originPatch) {
+            if (str.startsWith("diff -u ")) {
```
Comment (rdiachenko) [inline src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: It is not a part of a diff. Why do you use this information to split the files?

File: src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+diff -u Origin/Test1.java Update/Test1.java
```
Comment (rdiachenko) [inline src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt:1]: Why this line is a part of a patch?

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java [hunk 3]
```diff
@@ -74,9 +71,15 @@ private boolean isFileNameMatching(AuditEvent event) {
      * @return true if line and column are matching or not set.
      */
     private boolean isLineMatching(AuditEvent event) {
-        Boolean result = false;
+        boolean result = false;
         if (event.getLocalizedMessage() != null) {
-            result = startLine <= event.getLine() && event.getLine() <= endLine;
+            for (List<Integer> aLineRangeList : lineRangeList) {
+                result = aLineRangeList.get(0) <= event.getLine()
+                        && event.getLine() <= aLineRangeList.get(1);
```
Comment (rdiachenko) [inline src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java:78]: You are doing `lineRange.add(targetChunk.getPosition() + targetChunk.getLines().size());` it means `event.getLine() < aLineRangeList.get(1)` but not `event.getLine() <= aLineRangeList.get(1)`.
The problem here is that you may give violation on the line which was not changed. Please add a test for this case:
```
1 old line // violation 1
2 added line // violation 2
3 added line
4 added line // violation 3
5 old line // violation 4
```
Only violations 2 and 3 are expected. Here you cover few corner cases where violations are on the added code block edges.

File: src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt (hunk 2 above)
Comment (HuGanghui) [reply src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt:1]: when using diff to compare two directories, this line will be part of a patch

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: when using diff to compare two directories, this line will be part of a patch

Comment (rdiachenko) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: Please give example, I don't see this line is a part of a patch:
```
/var/tmp/checkstyle $ ls a b
a:
Origin.java
b:
Update.java

/var/tmp/checkstyle $ diff -u a/Origin.java b/Update.java 
--- a/Origin.java	2020-06-12 08:34:17.002841681 +0100
+++ b/Update.java	2020-06-12 08:34:26.576808407 +0100
@@ -6,6 +6,11 @@
         return;
       } else if (error > 40) {
         System.out.println("ok");
+        return;
+      } else if (error > 30) {
+        return;
+      } else {
+        System.out.println("pass");
       }
     }
 }
```

Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: This need more than one file in every directories.
Origin dir:

cat Origin/test1.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Origin;

public class Test1 {
    public void test1() {

    }
}

```
cat Origin/test2.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Origin;

public class Test2 {
    public void test1() {
        System.out.println();
    }
}
```
Update dir:

Update/test1.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Update;

public class Test1 {
    public void test1() {

    }

    public void test2() {

    }

    public void test3() {

    }
}

```
Update/test2.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Update;

public class Test2 {
    public void test1() {
        System.out.println();
        System.out.println();
        System.out.println();
    }
}
```
run diff command 

```sh
diff -u Origin/ Update/ > multiPatch.txt
```
cat multiPatch.txt
```
diff -u Origin/Test1.java Update/Test1.java
--- Origin/Test1.java	2020-06-11 10:48:40.000000000 +0800
+++ Update/Test1.java	2020-06-11 10:48:40.000000000 +0800
@@ -1,7 +1,15 @@
-package checkstyle_demo.PatchSuppression.multiFile.Origin;
+package checkstyle_demo.PatchSuppression.multiFile.Update;
 
 public class Test1 {
     public void test1() {
 
     }
+
+    public void test2() {
+
+    }
+
+    public void test3() {
+
+    }
 }
diff -u Origin/Test2.java Update/Test2.java
--- Origin/Test2.java	2020-06-11 10:48:40.000000000 +0800
+++ Update/Test2.java	2020-06-11 10:48:41.000000000 +0800
@@ -1,7 +1,9 @@
-package checkstyle_demo.PatchSuppression.multiFile.Origin;
+package checkstyle_demo.PatchSuppression.multiFile.Update;
 
 public class Test2 {
     public void test1() {
         System.out.println();
+        System.out.println();
+        System.out.println();
     }
 }

```

**Comment by romani** [comment]
@HuGanghui , please rebase , I merged your commit with organized inputs

**Comment by HuGanghui** [comment]
@romani finish rebase

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java (hunk 3 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java:78]: Yes, this may give violation on the line which was not changed, but there is no other way to deal with it when using java-diff-utils, as showing in follows

image

Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java:78]: But there is a way to avoid this problem. When using The unified output format of diff command, there is a option: `To select this output format, use the -U lines, --unified[=lines], or -u option. The argument lines is the number of lines of context to show. When it is not given, it defaults to three.`

File: src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt (hunk 2 above)
Comment (romani) [reply src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt:1]: Please show example if how it is part of patch. Terminal Command "diff -u ..." is not part of patch.

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: This is a nuances of diff tool that it prints command line for each file, it is not a case for patch files. Please open any git diff  to see hw it looks. https://github.blog/2011-10-21-github-secrets/#diff-patch

File: src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt (hunk 2 above)
Comment (HuGanghui) [reply src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt:1]: example is https://github.com/checkstyle/patch-filters/pull/13#discussion_r439289102, I mentioned above.

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java:123]: there is `diff --git` above each file - https://github.com/checkstyle/checkstyle/commit/c42a0e1b63c33400e05c98b62179cf303724e525.patch

```
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
index adae0fe2d0..46bd2bf257 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
```

File: src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt (hunk 2 above)
Comment (romani) [reply src/test/resources/com/github/checkstyle/patchfilter/MultiChangedFilesOnOnePatch.txt:1]: https://github.com/checkstyle/patch-filters/pull/13#discussion_r439742609 - proves that is one of possible cases.

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java (hunk 3 above)
Comment (romani) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java:78]: @HuGanghui , please use any value context you like, for POC phase we do not care that much.
you do not need to support all possible modes of diff now, choose one (the most easy to implement) and we will use it.

**Comment by romani** [comment]
@HuGanghui , @rdiachenko , reminder ....
we do not need well implemented functionality, all we need in POC is code that works with `git diff`, that we will use during testing.
All other we can do later on, if something is noticed and good to do but not requied for POC, please move to separate issue.

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java (hunk 3 above)
Comment (rdiachenko) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java:78]: @HuGanghui I don't understand the problem you have, here I meant to change `<=` onto `<`, please check it, this should solve the problem I described

> it means event.getLine() < aLineRangeList.get(1) but not event.getLine() <= aLineRangeList.get(1)

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java]: done

File: src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java (hunk 3 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterElement.java]: done, sorry I misunderstood at first
